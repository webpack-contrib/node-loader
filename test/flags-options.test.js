/**
 * @jest-environment node
 */

import os from 'os';

import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getModuleSource,
  getWarnings,
  readAsset,
} from './helpers';

describe('"flags" option', () => {
  it.only('should work', async () => {
    const compiler = getCompiler('simple.js', {
      flags: os.constants.dlopen.RTLD_NOW,
    });
    const stats = await compile(compiler);

    expect(getModuleSource('./hello_world.node', stats)).toMatchSnapshot(
      'module'
    );
    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getErrors(stats)).toMatchSnapshot('errors');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
  });
});
