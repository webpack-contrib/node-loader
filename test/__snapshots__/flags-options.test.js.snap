// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"flags" option should work: errors 1`] = `Array []`;

exports[`"flags" option should work: module 1`] = `
"
try {
  global.process.dlopen(module, \\"/absolute/path/to/test/fixtures/hello_world.node\\", 2);
} catch (error) {
  throw new Error('node-loader:/n' + error);
}
"
`;

exports[`"flags" option should work: result 1`] = `
Object {
  "default": Object {
    "sayHi": [Function],
  },
}
`;

exports[`"flags" option should work: warnings 1`] = `Array []`;
